app/src/components/toggleTip/toggleTip.test.tsx

```diff
@@ -0,0 +1,61 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { ToggleTip } from "./toggleTip";
+
+describe("ToggleTip", () => {
+  const text = "More info";
+  const toggleTipContent = "This is the tooltip content.";
+
+  it("renders button with an icon", () => {
+    render(<ToggleTip text={text} toggleTipContent={toggleTipContent} />);
+    expect(
+      screen.getByRole("button", { name: /more info/i })
+    ).toBeInTheDocument();
+  });
+
+  it("shows tooltip on hover and hides it on mouse leave", async () => {
+    render(<ToggleTip text={text} toggleTipContent={toggleTipContent} />);
+    const button = screen.getByRole("button");
+
+    fireEvent.mouseEnter(button);
+    expect(screen.getByText(toggleTipContent)).toBeInTheDocument();
+
+    fireEvent.mouseLeave(button);
+    expect(screen.queryByText(toggleTipContent)).not.toBeInTheDocument();
+  });
+
+  it("shows tooltip on focus and hides it on blur", async () => {
+    render(<ToggleTip text={text} toggleTipContent={toggleTipContent} />);
+    const button = screen.getByRole("button");
+
+    fireEvent.focus(button);
+    expect(screen.getByText(toggleTipContent)).toBeInTheDocument();
+
+    fireEvent.blur(button);
+    expect(screen.queryByText(toggleTipContent)).not.toBeInTheDocument();
+  });
+
+  it("toggles tooltip on click and requires another click to close", async () => {
```

Comment:
If this is strictly following the link from Clare, it should only open when it's clicked and close when it's clicked outside.